ably/proto/message.go

```diff
@@ -94,11 +314,19 @@ func (m *Message) DecodeData(opts *ChannelOptions) error {
 //
 // Any errors encountered in any step will be returned immediately.
 func (m *Message) EncodeData(encoding string, opts *ChannelOptions) error {
```

Comment:
What happens if `EncodeData` is called with the same parameters on the same message twice?